module5/RawParser.py

```diff
@@ -0,0 +1,43 @@
+import os.path
+import pathlib
+from typing import List, Dict
+
+from module4 import string_object_ref as sof
+from module5.InputException import InputException
+
+
+class RawParser:
+    DEFAULT_FILE_PATH = os.path.join(pathlib.Path().resolve(), 'landing', 'raw-publications.txt')
+
+    def __init__(self, file_path):
+        self.file_path = file_path
+
+    def parse_raw_file(self):
+        return self.__get_pub_objects(self.__read_file())
+
+    def __read_file(self) -> List[str]:
+        with open(self.file_path) as file:
+            lines = file.readlines()
+        return lines
+
+    def __get_pub_objects(self, lines: List[str]) -> List[Dict[str, str]]:
+        try:
+            publications = []
+            for line in lines:
+                objs = line.split(";")
+                pub = {'type': objs[0].lower().replace(" ", ""),
```

Comment:
maybe also apply sof?

Reply:
The sof module doesn't contain a suitable function for this transformation. The main purpose of this transformation is to cleanup any possible formatting and spaces for accurately identifying the type of publication in the file. The header text isn't taken from this file, it is formed in the code.

Reply:
I mean if you use functions from another module for cleaning input data (all of three properties are from file), let's implement functions for cleaning in those module and apply them in an abstract way. But it's up to you:)